### DISCUSSION TITLE
Hot Reload Not Working with WSL

### DISCUSSION THREAD

**Discussion by CubingSoda** [vercel/next.js#29159] [Help]
Hello,

I am on a Windows 10 machine, with WSL2 enabled. I have Ubuntu enabled on my system. 

I installed npm and yarn, then ran this:

`yarn create next-app test`

It finished successfully. 

However, after running `yarn dev` and making a change, it won't hot reload. It didn't even show my changes after manually hard refreshing. When I re-ran `yarn dev`, it updated the site. Why is this happening?

When I used Nextjs without WSL, it worked perfectly. 

Help greatly appreciated.

**Comment by AdsonCicilioti** [comment]
Yes, @BypassSpace ! this problem is already known and so far the best solution in my opinion is to change the port.

Something on port 3000 on WSL with HMR just doesn't work well. Edit `package.json` and in the `dev` scripts add `-p 4001` (for example).

Be happy for now!

**Reply by CubingSoda** [reply]
Hello @AdsonCicilioti, 

Thank you for your response. After setting the value of `dev` to `"next dev -p 4001"`, it still didn't work. The same problem still persisted.

**Reply by neonvoidx** [reply]
getting same issue, even with port changed

**Reply by rituraj-OO7** [reply]
even after changing the port. HMR is not working for me

**Reply by icyJoseph** [reply]
Did you guys ever try with file watch polling? if you are using webpack, like `WATCHPACK_POLLING=1000 next dev`
